File: 03 network/tcpip/echo_client.py
#!/usr/bin/env python
#-*- coding: utf-8 -*-
#If this code works, it was written by Seunghyun Lee(www.bluebaynetworks.co.kr).
#If not, I don't know who wrote it

import socket

#만약 통신할 에코 서버가 로컬이 아니면 아래 IP 주소를 알맞게 수정한다.
TCP_IP = '127.0.0.1'
#에코 테스트에 사용할 포트 번호. 에코 서버에서 bind 함수에서 사용한 값을 이용한다.
TCP_PORT = 5005
BUFFER_SIZE = 1024

#통신에 사용할 TCP/IP 소켓 생성
s = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
#타임 아웃을 5초로 설정
s.settimeout(5)
try:
	s.connect((TCP_IP, TCP_PORT))
except socket.error, exc:
	#접속 에러 발생시 에러코드 및 내용을 보여주고 프로그램을 종료
	print "Caught exception socket.error : %s" % exc
	exit(0)
try: 
	while True:
		#송신 데이터를 입력받음
		MESSAGE = raw_input("Input:")
		print "Send:", MESSAGE
		#에코서버에 데이터 송신
		sent = s.send(MESSAGE)
		if not sent: break
		#에코 서버가 보낸 데이터 수신
		data = s.recv(BUFFER_SIZE)
		if not data: break
		print "RECV:", data

except KeyboardInterrupt:   
	print "Now Exit"
finally:
	s.close()

print "socket disconnected. Exit"

File: 03 network/tcpip/echo_client.c
#include <sys/types.h>
#include <sys/socket.h>
#include <netdb.h>
#include <stdio.h>
#include<string.h>
 
int main(int argc,char **argv)
{
    int sockfd,n;
    char MESSAGE[1024];
    struct sockaddr_in servaddr;
 
    sockfd=socket(AF_INET,SOCK_STREAM,0);
    bzero(&servaddr,sizeof servaddr);
 
    servaddr.sin_family=AF_INET;
    servaddr.sin_port=htons(5005);
 
    inet_pton(AF_INET,"127.0.0.1",&(servaddr.sin_addr));
 
    connect(sockfd,(struct sockaddr *)&servaddr,sizeof(servaddr));
 
    while(1)
    {
		printf("Input:");
		gets(MESSAGE);
		printf("Send:%s\n", MESSAGE);
        n = write(sockfd, MESSAGE, strlen(MESSAGE)+1);
		if(0 >= n) break;
		memset(MESSAGE, 0x00, sizeof(MESSAGE));
        n = read(sockfd,MESSAGE,1024);
		if(0 >= n) break;
        printf("RECV:%s\n",MESSAGE);
    }
	 printf("socket disconnected. Exit\n");
	 close(sockfd);
}